py_gtfs_rt_ingestion/lib/convert_gtfs_rt.py

```diff
@@ -84,92 +105,129 @@ def convert(self) -> None:
                 if table.num_rows > 0:
                     table_count += 1
 
+                process_logger.add_metadata(table_count=table_count)
+
                 # limit number of tables produced on each event loop
                 if table_count >= max_tables_to_convert:
                     break
 
         except Exception as exception:
             process_logger.log_failure(exception)
         else:
-            process_logger.add_metadata(table_count=table_count)
             process_logger.log_complete()
 
+    def thread_init(self) -> None:
+        """
+        initialize the filesystem in each convert thread
+
+        update the active fs to use the s3 filesystem for all loading if the
+        first file starts with s3
+        """
+        thread_data = current_thread()
+        if self.files and self.files[0].startswith("s3://"):
+            thread_data.__dict__["file_system"] = fs.S3FileSystem()
+        else:
+            thread_data.__dict__["file_system"] = fs.LocalFileSystem()
+
```

Comment:
very similar to S3 object moving, initialize separate filesystem for each `convert` thread